From 3a6f40c5d0aa5095728fe6b83ba51d9bbedcbd45 Mon Sep 17 00:00:00 2001
From: Frank Li <Frank.Li@nxp.com>
Date: Tue, 8 May 2018 10:58:50 -0500
Subject: [PATCH 3741/5242] MLK-18237-3 XRDC: JPEG ENC/DEC fix crash missed
 power resource

commit  cca5f8a446ea463b7d9c686331a0c314f4a81a9c from
https://source.codeaurora.org/external/imx/linux-imx.git

[    5.184399] Unhandled fault: synchronous external abort (0x96000210) at 0xffff00000b000000
[    5.184824] (null): mxc_isi_capture_open, No remote pad found!
[    5.187470] (null): mxc_isi_capture_open, No remote pad found!
[    5.192734] (null): mxc_isi_capture_open, No remote pad found!
[    5.199931] (null): mxc_isi_capture_open, No remote pad found!
[    5.219447] Internal error: : 96000210 [#1] PREEMPT SMP
[    5.224681] Modules linked in:
[    5.227755] CPU: 2 PID: 3028 Comm: v4l_id Not tainted 4.9.88-04903-ga209cd8 #464
[    5.235162] Hardware name: Freescale i.MX8QXP MEK (DT)
[    5.240305] task: ffff80083411cb00 task.stack: ffff80083b7ac000
[    5.246254] PC is at clk_gate2_scu_enable+0x3c/0xa8

Signed-off-by: Frank Li <Frank.Li@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi |   24 ++++++++++++++++++++----
 1 file changed, 20 insertions(+), 4 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
index 3bd59869..e93c278 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
@@ -967,16 +967,32 @@
 				power-domains =<&pd_isi_ch0>;
 			};
 
-			pd_jpgdec: PD_IMAGING_JPEG_DEC {
-				reg = <SC_R_MJPEG_DEC_S0>;
+			pd_jpeg_dec_mp: PD_JPEG_DEC_MP{
+				reg = <SC_R_MJPEG_DEC_MP>;
 				#power-domain-cells = <0>;
 				power-domains =<&pd_isi_ch0>;
+				#address-cells = <1>;
+				#size-cells = <0>;
+
+				pd_jpgdec: PD_IMAGING_JPEG_DEC {
+					reg = <SC_R_MJPEG_DEC_S0>;
+					#power-domain-cells = <0>;
+					power-domains =<&pd_jpeg_dec_mp>;
+				};
 			};
 
-			pd_jpgenc: PD_IMAGING_JPEG_ENC {
-				reg = <SC_R_MJPEG_ENC_S0>;
+			pd_jpeg_enc_mp: PD_JPEG_ENC_MP{
+				reg = <SC_R_MJPEG_ENC_MP>;
 				#power-domain-cells = <0>;
 				power-domains =<&pd_isi_ch0>;
+				#address-cells = <1>;
+				#size-cells = <0>;
+
+				pd_jpgenc: PD_IMAGING_JPEG_ENC {
+					reg = <SC_R_MJPEG_ENC_S0>;
+					#power-domain-cells = <0>;
+					power-domains =<&pd_jpeg_enc_mp>;
+				};
 			};
 		};
 		pd_caam: PD_CAAM {
-- 
1.7.9.5

